Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -484,87 +567,169 @@ namespace AZ::Reflection
-                                m_visitor->VisitObjectEnd(*this, *this);
-                                m_stack.pop_back();
+                                groupPair.second.value().m_group = groupPair.first;
+                                m_stack.push_back(groupPair.second.value());
+                                CacheAttributes();
+                                m_visitor->VisitObjectBegin(*this, *this);
                             }
-                            if (!groupName.empty())
+
+                            for (const auto& groupEntry : nodeData.m_groupEntries[groupPair.first])
                             {
-                                // If our group has a name, we should actually call VisitObjectEnd, so continue with the group set
-                                m_stack.push_back(iter->second);
-                                m_stack.back().m_group = groupName;
+                                m_stack.push_back(groupEntry);
                                 CacheAttributes();
                                 m_visitor->VisitObjectBegin(*this, *this);
+                                m_visitor->VisitObjectEnd(*this, *this);
+                                m_stack.pop_back();
+                            }
+
+                            if (groupPair.second.has_value())
+                            {
+                                m_visitor->VisitObjectEnd(*this, *this);
+                                m_stack.pop_back();
                             }
                         }
-                        else if (iter->second.m_classElement->m_editData->m_elementId == AZ::Edit::ClassElements::UIElement)
+
+                        nodeData.m_iteratingOnGroups = false;
+                        nodeData.m_propertyToGroupMap.clear();
+                        nodeData.m_groupEntries.clear();
+                        nodeData.m_groups.clear();
+                    }
+
+                    m_stack.pop_back();
+                }
+
+                // The back of the stack now holds the parent.
+                {
+                    StackEntry& parentData = m_stack.back();
+
+                    if (!parentData.m_group.empty())
+                    {
+                        EndNode();
+                    }
+
+                    // Iterate over non serialized elements to see if any of them should be added
+                    const char* elementName = "";
+                    if (parentData.m_classElement && parentData.m_classElement->m_editData && parentData.m_classElement->m_editData->m_name)
+                    {
+                        elementName = parentData.m_classElement->m_editData->m_name;
+                    }
+                    auto iter = m_nonSerializedElements.begin();
+                    while (iter != m_nonSerializedElements.end())
+                    {
+                        // TODO - Fix this
```

Comment:
github really mangled this diff, but as far as I can tell, this is a new todo. Could you file a GHI for this? We've been asked to file GHIs for any known todos to help the community track this stuff...

Reply:
Hey, no this TODO already existed. I'll check if it was already tracked and add a GHI if that's not the case.

Reply:
oh, no worries then. I just didn't see it in the diff